tests/integration/Owncloud/OcisPhpSdk/DriveTest.php

```diff
@@ -2,17 +2,70 @@
 
 namespace integration\Owncloud\OcisPhpSdk;
 
+use Owncloud\OcisPhpSdk\Drive;
+use Owncloud\OcisPhpSdk\Exception\BadRequestException;
+use Owncloud\OcisPhpSdk\Exception\NotFoundException;
+use Owncloud\OcisPhpSdk\Ocis;
+
 require_once __DIR__ . '/OcisPhpSdkTestCase.php';
 
 class DriveTest extends OcisPhpSdkTestCase
 {
+    private Drive $drive;
+    private Ocis $ocis;
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->ocis = $this->getOcis('admin', 'admin');
+        $this->drive = $this->ocis->createDrive('test drive');
+        $this->createdDrives[] = $this->drive->getId();
+    }
+
     public function testDisableDrive(): void
     {
-        $ocis = $this->getOcis('admin', 'admin');
-        $drive = $ocis->createDrive('test drive');
-        $this->createdDrives[] = $drive->getId();
-        $this->assertFalse($drive->isDisabled());
-        $drive->disable();
-        $this->assertTrue($drive->isDisabled());
+        $this->assertFalse($this->drive->isDisabled());
+        $this->drive->disable();
+        $this->assertTrue($this->drive->isDisabled());
+    }
+
+    public function testEnableDrive(): void
+    {
+        $this->drive->disable();
+        $this->drive->enable();
+        $this->assertFalse($this->drive->isDisabled());
+    }
+
+    public function testEnableNotExistingDrive(): void
+    {
+        $this->expectException(NotFoundException::class);
+        $this->drive->disable();
+        $this->drive->delete();
+        $this->drive->enable();
+        $this->assertFalse($this->drive->isDisabled());
+    }
+
+    public function testDeleteDrive(): void
+    {
+        $this->drive->disable();
+        $this->drive->delete();
+        $this->expectException(NotFoundException::class);
+        $this->expectExceptionMessage('itemNotFound - no drive returned from storage');
+        $this->ocis->getDriveById($this->drive->getId());
```

Comment:
don't we have to remove id from $this->createdDrives[] after delete drive

Reply:
we do not update the object itself

Reply:
That is a more general question: #110